apps/common/src/main/scala/org/lfdecentralizedtrust/splice/store/db/DbMultiDomainAcsStore.scala

```diff
@@ -730,6 +739,50 @@ final class DbMultiDomainAcsStore[TXE](
     }
   }
 
+  override def listInterfaceViews[C, ICid <: ContractId[?], View <: DamlRecord[View]](
+      companion: C,
+      limit: Limit,
+  )(implicit
+      companionClass: ContractCompanion[C, ICid, View],
+      tc: TraceContext,
+  ): Future[Seq[Contract[ICid, View]]] = waitUntilAcsIngested {
+    val interfaceId = companionClass.typeId(companion)
+    val opName = s"listInterfaceViews:${interfaceId.getEntityName}"
+    for {
+      // assumption: most rows will satisfy `AND interface_view IS NOT NULL`, so we don't need to index for it
+      rows <- storage.query(
+        sql"""
+             SELECT contract_id, interface_view, acs.created_at, acs.created_event_blob
+             FROM #$interfaceViewsTableName interface
+               JOIN #$acsTableName acs ON acs.event_number = interface.acs_event_number
+             WHERE interface_id_qualified_name = ${QualifiedName(interfaceId)}
+               AND interface_view IS NOT NULL
+             ORDER BY interface.acs_event_number
+           """.as[(String, Json, Timestamp, Array[Byte])],
+        opName,
+      )
+    } yield {
+      val limited = applyLimit(opName, limit, rows)
+      limited.map { case (contractId, viewJson, createdAt, createdEventBlob) =>
+        companionClass
+          .fromJson(companion)(
+            interfaceId,
+            contractId,
+            viewJson,
+            ByteString.copyFrom(createdEventBlob),
+            createdAt.toInstant,
+          )
+          .fold(
+            err =>
+              throw new IllegalStateException(
+                s"Stored a contract that cannot be decoded as interface view $interfaceId: $err"
```

Comment:
Let's make debugging easier by also printing a unique identifier for the data that could not be decoded? Or potentially even the data itself?